### DISCUSSION TITLE
Hide specific node from sitemap

### DISCUSSION THREAD

**Discussion by eliranrefael** [patrickdemooij9/SeoToolkit.Umbraco#173] [Q&A]
hey,
is there anyway i can do it on the node level?

thanks!

**Comment by patrickdemooij9** [comment]
Hi @eliranrefael 
There is no way to do that with the interface currently. It is on my list of things to add to the package. Though there is a way to do this from within the code. You can hook onto the GenerateSitemapNotification and remove the nodes from there (https://seotoolkit.gitbook.io/useotoolkit/extensions/notifications#generatesitemapnotification).

**Reply by eliranrefael** [reply]
hey thanks for answering,
im planning to try what u suggested. if i understand correctly, i need to implement a switch, and scan all nodes, and remove the ones that has this switch one.
it sounds like a heavy task for the server, the question is how often this notification occurs

**Reply by eliranrefael** [reply]
hey, working with this code snippet, thanks alot, hope it wont be heavy on my prod.

![image](https://user-images.githubusercontent.com/37138835/225569837-a0f9527b-635c-44c8-921c-db8481cda2cd.png)

**Reply by patrickdemooij9** [reply]
The notification will only trigger once. So depending on the amount of nodes, it could be a bit slow. I'll see if I can find another faster way to do this.

For now, your code snippet will indeed do the trick.

**Comment by eliranrefael** [comment]
Once in a request? When site map requested?

My fear is that in a situation of 1000 nodes it will end in time out...

**Reply by patrickdemooij9** [reply]
Yeah, it is once in a request. I'll see if I can improve the logic on this

**Comment by eliranrefael** [comment]
I suggest regardless to my issue to add a setting for the frequency needed
for the xml generation, most sites won't need it to be generated more than
once a day, the live updating of this xml isn't critical...
